packages/kotti-table/src/logic/column.js

```diff
@@ -17,25 +17,40 @@ export const defaultState = {
 export const mutations = {
 	insertColumn(store, { column, index }) {
 		const deleted = store.state._destroyedColumns[column.prop]
-		setColumn(store.state, { column, index, deleted })
+		setColumn(store.state, {
+			column,
+			index,
+			deleted,
+		})
 		deleted && (store.state._destroyedColumns[column.prop] = 0)
-		store.commit('updateColumns', { emitChange: false })
+		store.commit('updateColumns', {
+			emitChange: false,
+		})
 	},
 	removeColumn(store, column) {
 		column = getColumn(store.state, column)
 		if (column) {
 			column._deleted = true
 			store.state._destroyedColumns[column.prop] = 1
-			store.commit('updateColumns', { emitChange: false })
+			store.commit('updateColumns', {
+				emitChange: false,
+			})
 		}
 	},
 	setColumns(store, columns = store.state._columnsArray) {
 		const { state } = store
 		const pickedColumns = columns.map(col => pick(col, PUBLIC_COLUMN_PROPS))
-		for (const col of pickedColumns) {
+		for (let i = 0; i < pickedColumns.length; i++) {
+			const col = pickedColumns[i]
 			const column = getColumn(state, col)
-			if (column) {
+
+			if (column && state._columns.length > 0) {
```

Comment:
_columns is an object dictionary it has no length
_columnsArray is the array form of _columns

also there's no need to do that check as getColumn will return undefined if a column is not in _columns

and there for you can just return the condition to `if (colum) { ... } else { ... }`